sections/shared/Layout/AppLayout/Header/BalanceActions.tsx

```diff
@@ -1,250 +1,275 @@
+		[kwentaOpRewards, snxOpRewards, tradingRewards]
 	);
 
-	const options = useMemo(() => {
-		const isolatedPositionsFiltered = isolatedPositions
-			.filter((position) => position.remainingMargin.gt(zeroBN))
-			.map((position) => setMarketConfig(position, FuturesAccountTypes.ISOLATED_MARGIN));
-		const crossPositionsFiltered = crossPositions
-			.filter((position) => position.remainingMargin.gt(zeroBN))
-			.map((position) => setMarketConfig(position, FuturesAccountTypes.CROSS_MARGIN));
-		return [
-			{
-				label: 'header.balance.total-margin-label',
-				totalAvailableMargin: formatDollars(portfolio.total),
-				options: [...isolatedPositionsFiltered, ...crossPositionsFiltered],
-			},
-		];
-	}, [crossPositions, isolatedPositions, setMarketConfig, portfolio]);
-
-	const OptionsGroupLabel: FC<{ label: string; totalAvailableMargin?: string }> = ({
-		label,
-		totalAvailableMargin,
-	}) => (
-		<FlexDivRow>
-			<Container>{t(label)}</Container>
-			<Container>{totalAvailableMargin}</Container>
-		</FlexDivRow>
-	);
+	const claimDisabledKwentaOp = useMemo(() => kwentaOpRewards.lte(0), [kwentaOpRewards]);
 
-	const formatOptionLabel: FC<ReactSelectOptionProps> = ({
-		label,
-		synthIcon,
-		marketRemainingMargin,
-		onClick,
-	}: ReactSelectOptionProps) => (
-		<LabelContainer onClick={onClick}>
-			<FlexDivRow>
-				{synthIcon && <StyledCurrencyIcon currencyKey={synthIcon} width="24px" height="24px" />}
-				<StyledLabel>{t(label)}</StyledLabel>
-			</FlexDivRow>
-			<Container>{marketRemainingMargin}</Container>
-		</LabelContainer>
-	);
+	const claimDisabledSnxOp = useMemo(() => snxOpRewards.lte(0), [snxOpRewards]);
+
+	const REWARDS = [
+		{
+			key: 'trading-rewards',
+			title: t('dashboard.rewards.trading-rewards.title'),
+			copy: t('dashboard.rewards.trading-rewards.copy'),
+			button: t('dashboard.rewards.staking'),
+			kwentaIcon: true,
+			linkIcon: true,
+			rewards: tradingRewards,
+			onClick: goToStaking,
+			isDisabled: false,
+		},
+		{
+			key: 'kwenta-rewards',
+			title: t('dashboard.rewards.kwenta-rewards.title'),
+			copy: t('dashboard.rewards.kwenta-rewards.copy'),
+			button: t('dashboard.rewards.claim'),
+			kwentaIcon: false,
+			linkIcon: false,
+			rewards: kwentaOpRewards,
+			onClick: handleClaimOp,
+			isDisabled: claimDisabledKwentaOp,
+		},
+		{
+			key: 'snx-rewards',
+			title: t('dashboard.rewards.snx-rewards.title'),
+			copy: t('dashboard.rewards.snx-rewards.copy'),
+			button: t('dashboard.rewards.claim'),
+			kwentaIcon: false,
+			linkIcon: false,
+			rewards: snxOpRewards,
+			onClick: () => {},
+			isDisabled: claimDisabledSnxOp,
+		},
+	];
 
 	useEffect(() => {
-		setBalanceLabel(formatDollars(susdWalletBalance, { sign: '$' }));
-	}, [balanceLabel, susdWalletBalance]);
+		const tokenAddresses = [KWENTA_ADDRESS, OP_ADDRESS];
+		const initExchangeTokens = async () => {
+			const coinGeckoPrices = await sdk.exchange.batchGetCoingeckoPrices(tokenAddresses, true);
```

Comment:
async calls should be wrapped in a try catch, as this would result in an unhandled error. But also I think we can just add OP to `getEarnTokenPrices` and use the `fetchEarnTokenPrices` action. Ideally we shouldn't be making async requests inside components but keeping this sort of data fetching and storage in redux / sdk.

Reply:
Thatnks. That's a better solution. Will change this.